src/main/java/teammates/ui/webapi/action/GetStudentAction.java

```diff
@@ -17,37 +25,65 @@ protected AuthType getMinAuthLevel() {
-            StudentAttributes studentAttributes = getStudentOfCourseFromRequest(feedbackSession.getCourseId());
-            checkAccessControlForStudentFeedbackSubmission(studentAttributes, feedbackSession);
-            break;
-        case FULL_DETAIL:
-            // TODO implement this when necessary
-            break;
-        default:
-            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        CourseAttributes course = logic.getCourse(courseId);
+
+        StudentAttributes student = null;
+
+        String studentEmail = getRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+        String regKey = getRequestParamValue(Const.ParamsNames.REGKEY);
+
+        if (studentEmail != null) {
+            student = logic.getStudentForEmail(courseId, studentEmail);
+            if (student == null || !userInfo.isInstructor) {
+                throw new UnauthorizedAccessException(UNAUTHORIZED_ACCESS);
+            }
+
+            InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+            gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId), student.section,
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS);
+        } else if (regKey != null) {
+            getUnregisteredStudent().orElseThrow(() -> new UnauthorizedAccessException(UNAUTHORIZED_ACCESS));
+        } else {
+            if (!userInfo.isStudent) {
+                throw new UnauthorizedAccessException(UNAUTHORIZED_ACCESS);
+            }
+
+            student = logic.getStudentForGoogleId(courseId, userInfo.id);
+            gateKeeper.verifyAccessible(student, course);
         }
     }
 
     @Override
     public ActionResult execute() {
-        Intent intent = Intent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
-        switch (intent) {
-        case STUDENT_SUBMISSION:
-            String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-            StudentAttributes studentAttributes = getStudentOfCourseFromRequest(courseId);
-            return new JsonResult(new StudentInfo.StudentResponse(studentAttributes));
-        case FULL_DETAIL:
-            // TODO implement this when necessary
-            return null;
-        default:
-            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        StudentAttributes student = null;
+
+        String studentEmail = getRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+        String regKey = getRequestParamValue(Const.ParamsNames.REGKEY);
+
+        if (studentEmail != null) {
+            student = logic.getStudentForEmail(courseId, studentEmail);
+        } else if (regKey != null) {
+            Optional<StudentAttributes> studentCheck = getUnregisteredStudent();
```

Comment:
Consider using Optional#ifPresent

Reply:
I have realised a few things (while investigating coverage) and refactored this block